graphql/src/graphql.ml

```diff
@@ -1,6 +1,15 @@
-module Schema = Graphql_schema.Make(struct
+module Io = struct
   type +'a t = 'a
 
   let bind x f = f x
   let return x = x
-end)
\ No newline at end of file
+end
+
+module Stream = struct
+  type +'a io = 'a Io.t
+  type 'a t = 'a list
```

Comment:
Maybe it's silly to spend too much timing getting this type right, as a pubsub stream doesn't make much too much sense in a synchronous setting. However, after thinking more about it, I think `list` is a poor fit (though I did suggest it myself to begin with -- sorry about that). After looking more into `Stream`, `Seq` and `Queue`, I think `Seq` is the best option. Even though `Stream` is used as the stream type in [`ocaml-redis`](https://github.com/0xffea/ocaml-redis/blob/9a5c0ba79fc714ce719061349fd4457d9f119828/src_sync/redis_sync.ml#L12), it seems like it may `Stream` [may be deprecated](https://caml.inria.fr/mantis/view.php?id=6711). Note that this will require adding a dependency on `seq` for OCaml versions before 4.07.0.

Reply:
Yeah, a `Seq` sounds appropriate, but I don't know how to add conditional dependencies to either Dune or Opam to only build on targets less than 4.07. Little help?